app/src/modules/content/routes/item.vue

```diff
@@ -352,20 +352,30 @@ async function saveVersionAction(action: 'main' | 'stay' | 'quit') {
 				refresh();
 				break;
 			case 'quit':
-				if (!props.singleton) router.push(`/content/${props.collection}`);
+				redirectAfterSave();
 				break;
 		}
 	} catch {
 		// Save version shows unexpected error dialog
 	}
 }
 
+function redirectAfterSave() {
+	const returnToPreviousPage = router.currentRoute.value.query?.return;
+
+	if (returnToPreviousPage === 'true' && !props.singleton) {
+		router.back();
```

Comment:
Lets use the `navigateBack()` function in this component to handle going back so it gracefully handles times where the back page may not be available. https://github.com/directus/directus/blob/e99b4f9593ad2d32521d29e60f8a1b03c3d01540/app/src/modules/content/routes/item.vue#L317